MINOR: log AssignReplicasToDirsRequest dispatch and handling

gaurav-narula opened the pull request:
It's hard to ascertain at the moment if AssignmentsManager dispatched an `AssignReplicasToDirsRequest` to the controller and if the controller handled it. This PR adds info logs for observability.

gaurav-narula committed:
MINOR: log AssignReplicasToDirsRequest dispatch and handling

It's hard to ascertain at the moment if AssignmentsManager dispatched
an `AssignReplicasToDirsRequest` to the controller and if the controller
handled it. This PR adds info logs for observability.

OmniaGM commented on server/src/main/java/org/apache/kafka/server/AssignmentsManager.java:
```diff
@@ -270,8 +270,8 @@ public void run() throws Exception {
                     .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().dirId));
-            if (log.isDebugEnabled()) {
-                log.debug("Dispatching {} assignments:  {}", assignment.size(), assignment);
+            if (log.isInfoEnabled()) {
```
Is `log.isInfoEnabled` necessary here? I think we need this log in any log level

gaurav-narula replied:
I adhered to the same convention as was before. It turns out the checks are useful when doing string concatenation instead of parameterised messages [0](https://www.slf4j.org/faq.html#logging_performance).

Since we're using parameterised messages, I think we can safely avoid the conditionals. Adding a commit for the same. 👍

OmniaGM reviewed (commented):
The pr seems straightforward just left one comment

gaurav-narula committed:
Avoid conditional

We can avoid it without much perf cost since the logs use parameterised
messaging

OmniaGM reviewed (approved):
Thanks for addressing the comment. LGTM now

pprovenzano reviewed (approved):
LGTM
